feg/gateway/services/n7_n40_proxy/n7_n40_proxy/main.go

```diff
@@ -37,21 +39,37 @@ func main() {
 		glog.Fatalf("Error creating N7_N40 Proxy service: %s", err)
 	}
 
+	n7config, err := n7.GetN7Config()
+	if err != nil {
+		glog.Fatalf("Error fetching config: %s", err)
+	}
 	cloudReg := registry.Get()
 	dbClient, err := policydb.NewRedisPolicyDBClient(cloudReg)
 	if err != nil {
 		glog.Fatalf("Error connecting to redis store from N7_N40 Proxy: %s", err)
 	}
-
-	sessController, err := servicers.NewCentralSessionController(dbClient)
+	policyClient, err := n7.NewN7Client(&n7config.Server)
+	if err != nil {
+		glog.Fatalf("Creating N7 Client failed: %s", err)
+	}
+	sessController, err := servicers.NewCentralSessionController(n7config, dbClient, policyClient)
 	if err != nil {
 		glog.Fatalf("Error creating session controller in N7_N40 Proxy: %s", err)
 	}
+	notifHandler := notify.NewNotificationHandler(&n7config.Client, cloudReg)
+	if notifHandler == nil {
+		glog.Fatalf("Error creating noitification handler")
+	}
 
 	// Add GRPC handlers to the service
 	lteprotos.RegisterCentralSessionControllerServer(srv.GrpcServer, sessController)
 	protos.RegisterServiceHealthServer(srv.GrpcServer, sessController)
 
+	// Run the Notification Handler HTTP service
+	_, err = notifHandler.Start()
+	if err != nil {
+		glog.Fatalf("Failed to start Notification Handler")
+	}
```

Comment:
same here. I think this should be part of the n7 proxy.

Or is will this server be shared with the n40?

[here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Fservices%2Fsession_proxy%2Fcredit_control%2Fgx%2Fgx_client.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=86&start_col=23&end_row=86&end_col=23) is how one of the hanlders is added on session proxy. As you can see it is a little bit of a mess

Reply:
N40 will also use the same notification handler.

NotificationHandler can be moved to session_controller.go. However in that case for create, delete or update tests (example create_session_test.go) the NotificationHandler will be started and has to be Closed after the test, even though it is not used. Will this be okay?

Reply:
Adding a defer on the test is the right way better than modifying the design fo the test
If you are concerned about routines left behind you can check this

`"go.uber.org/goleak"`

https://medium.com/analytics-vidhya/goroutines-unit-testing-handling-unexpected-goroutines-dd0003be193c